pkg/cephfs/cephuser.go

```diff
@@ -53,83 +48,47 @@ func getCephUserName(volID volumeID) string {
 	return cephUserPrefix + string(volID)
 }
 
-func getCephUser(volOptions *volumeOptions, adminCr *credentials, volID volumeID) (*cephEntity, error) {
-	entityName := cephEntityClientPrefix + getCephUserName(volID)
-
+func getSingleCephEntity(args ...string) (*cephEntity, error) {
 	var ents []cephEntity
-	args := [...]string{
-		"-m", volOptions.Monitors,
-		"auth", "-f", "json", "-c", cephConfigPath, "-n", cephEntityClientPrefix + adminCr.id, "--keyring", getCephKeyringPath(volID, adminCr.id),
-		"get", entityName,
-	}
-
-	out, err := execCommand("ceph", args[:]...)
-	if err != nil {
-		return nil, fmt.Errorf("cephfs: ceph failed with following error: %s\ncephfs: ceph output: %s", err, out)
-	}
-
-	// Workaround for output from `ceph auth get`
-	// Contains non-json data: "exported keyring for ENTITY\n\n"
-	offset := bytes.Index(out, []byte("[{"))
-
-	if err = json.NewDecoder(bytes.NewReader(out[offset:])).Decode(&ents); err != nil {
-		return nil, fmt.Errorf("failed to decode json: %v", err)
+	if err := execCommandJSON(&ents, "ceph", args...); err != nil {
+		return nil, err
 	}
 
 	if len(ents) != 1 {
-		return nil, fmt.Errorf("got unexpected number of entities for %s: expected 1, got %d", entityName, len(ents))
+		return nil, fmt.Errorf("got unexpected number of entities: expected 1, got %d", len(ents))
 	}
 
 	return &ents[0], nil
 }
 
-func createCephUser(volOptions *volumeOptions, adminCr *credentials, volID volumeID) (*cephEntity, error) {
-	caps := cephEntityCaps{
-		Mds: fmt.Sprintf("allow rw path=%s", getVolumeRootPathCeph(volID)),
-		Mon: "allow r",
-		Osd: fmt.Sprintf("allow rw pool=%s namespace=%s", volOptions.Pool, getVolumeNamespace(volID)),
-	}
-
-	var ents []cephEntity
-	args := [...]string{
+func getCephUser(volOptions *volumeOptions, adminCr *credentials, volID volumeID) (*cephEntity, error) {
+	return getSingleCephEntity(
 		"-m", volOptions.Monitors,
-		"auth", "-f", "json", "-c", cephConfigPath, "-n", cephEntityClientPrefix + adminCr.id, "--keyring", getCephKeyringPath(volID, adminCr.id),
-		"get-or-create", cephEntityClientPrefix + getCephUserName(volID),
-		"mds", caps.Mds,
-		"mon", caps.Mon,
-		"osd", caps.Osd,
-	}
-
-	if err := execCommandJSON(&ents, args[:]...); err != nil {
-		return nil, fmt.Errorf("error creating ceph user: %v", err)
-	}
+		"-n", cephEntityClientPrefix+adminCr.id, "--key="+adminCr.key,
+		"-c", cephConfigPath,
+		"-f", "json",
+		"auth", "get", cephEntityClientPrefix+getCephUserName(volID),
```

Comment:
```suggestion
		"auth",
		"get", cephEntityClientPrefix+getCephUserName(volID),
```

Reply:
The aim here was to have related arguments on the same line (`auth get ENTITY-NAME` is a single "unit"), the same goes for credentials - `-n ENTITY-NAME --key=SECRET` is a single unit...but I'm fine with either if it looks too congested.